python_modules/event-pipeline-demo/event-pipeline-demo/solids.py

```diff
@@ -0,0 +1,97 @@
+            fields={
+                'deploy_target': Field(
+                    SparkDeployTarget,
+                    description='''This solid supports several deploy targets for your Spark 
+                    application: Local Spark (default), a remote Spark master, AWS EMR, and GCP 
+                    Dataproc.''',
+                    default_value=SparkDeployTargetLocal,
+                    is_optional=False,
+                ),
+                'main_class': Field(
+                    String,
+                    description='''The entry point for your application (e.g. 
+                    org.apache.spark.examples.SparkPi)''',
+                    is_optional=False,
+                ),
+                'master_url': Field(
+                    String,
+                    description='The master URL for the cluster (e.g. spark://23.195.26.187:7077)',
+                    is_optional=False,
+                ),
+                'deploy_mode': Field(
+                    SparkDeployMode,
+                    description='''Whether to deploy your driver on the worker nodes (cluster) or 
+                    locally as an external client (client) (default: client). A common deployment 
+                    strategy is to submit your application from a gateway machine that is physically
+                    co-located with your worker machines (e.g. Master node in a standalone EC2
+                    cluster). In this setup, client mode is appropriate. In client mode, the driver
+                    is launched directly within the spark-submit process which acts as a client to 
+                    the cluster. The input and output of the application is attached to the console.
+                    Thus, this mode is especially suitable for applications that involve the REPL 
+                    (e.g. Spark shell).''',
+                    default_value=SparkDeployModeClient,
+                    is_optional=False,
+                ),
+                'application_jar': Field(
+                    String,
+                    description='''Path to a bundled jar including your application and all 
+                    dependencies. The URL must be globally visible inside of your cluster, for 
+                    instance, an hdfs:// path or a file:// path that is present on all nodes.''',
+                    is_optional=False,
+                ),
+                'conf': Field(
+                    Dict(fields={}),
+                    description='''Arbitrary Spark configuration property in key=value format. For 
+                    values that contain spaces wrap “key=value” in quotes.''',
+                ),
+                'application_arguments': Field(
+                    Dict(fields={}),
+                    description='Arguments passed to the main method of your main class, if any',
+                ),
+            }
+        )
+    ),
+)
+def SparkJobSolid(context):
```

Comment:
yeah, that's why I switched back to `def define_spark_solid()` - but I think it's worth thinking about this interface some more, because once we get the core set of 10-20 "framework" solids in place, I think this will be the primary way that people interact with the system. In Airflow we'd often have:

```
a = SparkJob(...some config)
b = SomeOtherCustomOperator(....some config)
a.set_upstream(b)
```
which feels cleaner than the mixed :
```
a = define_spark_solid(...)
b= SolidDefinition('some other custom solid', ...)
...
```